support toggling DNSSEC on zones

rupa opened the pull request:
and a bit of whitespace cleanup in string literals in zone_test.go

Commit:
support toggling DNSSEC on zones

and a bit of whitespace cleanup in string literals in zone_test.go

rupa commented:
I did test error handling with DNSSEC not enabled on account, we do get good clear error message about that.

mburtless commented on rest/model/dns/zone.go:
```diff
@@ -37,6 +37,9 @@ type Zone struct {
 	Secondary *ZoneSecondary `json:"secondary,omitempty"`
+
+	// Whether or not DNSSEC is enabled on the zone
+	DNSSEC bool `json:"dnssec"`
```
This effectively makes `"dnssec": false` the default for all newly created zones via the SDK.  Not strictly a problem for the SDK, but could create issues downstream.  I.e. in Terraform the `resourceZoneUpdate` function in our provider constructs the body for the update request via the `NewZone` function in the SDK.

If the tf provider is updated to use this version of the SDK and is used to manage an existing tfstate on a zone with dnssec enabled, it will disable dnssec on the next update via TF (since no `dnssec` attribute will be included in the tf config and `"dnssec": false` will be included in the body by the sdk).

Maybe this can be worked around when adding the dnssec attribute to the tf provider.  Might be cleaner to implement `omitempty` on this attribute by changing the type to `*bool`?

rupa replied:
Good catch and nice way of solving it. This should be all set now.

Commit:
Use *bool for zone.DNSSEC

Preserves the "optional" quality of this attribute, so leaving it
unset won't change an existing setting.

mburtless reviewed (approved):
LGTM
